src/mpid/ch4/src/ch4r_callbacks.c

```diff
@@ -554,6 +554,10 @@ int MPIDIG_send_long_req_target_msg_cb(int handler_id, void *am_hdr, void **data
         MPID_THREAD_CS_EXIT(VCI, MPIDIU_THREAD_MPIDIG_GLOBAL_MUTEX);
     } else {
         /* Matching receive was posted, tell the netmod */
+        /* we still need hold onto comm for sending ack.
+         * This shouldn't be necessary if we use global rank for active messages */
+        MPIR_Comm_add_ref(root_comm);
```

Comment:
Strike the `MPIR_Comm_release` below and this add should not be needed.

Reply:
True. But the add_ref and the release are for different semantic ownership though. I guess replace both with an explanation comment will do.

Reply:
Hmm, yes I see your point. Backing up, I'm not sure why the other reference is released here. The message should be tied to the recv request, which is not yet complete. I agree that requiring comm to send the ack is not necessary in the long run.